src/index.rs

```diff
@@ -245,7 +248,20 @@ impl Iter {
         }
         self.report.increment_duration("index.parse", start_parse.elapsed());
 
-        self.buffered_entries = entries.into_iter();
+        self.buffered_entries = entries.into_iter()
+            .filter(|entry| {
+                if self.excludes.is_match(&entry.apath) {
+                    match entry.kind {
+                        IndexKind::Dir => self.report.increment("skipped.excluded.directories", 1),
+                        IndexKind::Symlink => self.report.increment("skipped.excluded.directories", 1),
+                        IndexKind::File => self.report.increment("skipped.excluded.files", 1)
+                    }
+                    return false;
+                }
+                return true;
+            })
+            .collect::<Vec<Entry>>().into_iter();
```

Comment:
I don't understand why you collect it and then make an iter from that?

Reply:
The result of `buffered_entries` has to be of type `vec::IntoIter<Entry>`. I have an `Vec<Entry>`and want to exclude some entries. Is there a better way to do that?

Reply:
Oh OK. Perhaps we could later refactor `buffered_entries` but what you have here seems fine.